internal/util/setters/setters.go

```diff
@@ -96,48 +95,69 @@ func PerformSetters(path string) error {
 			continue
 		}
 
-		setter := &setters.PerformSetters{
-			Name:  fmt.Sprintf("gcloud.%s", c),
-			Value: v,
-			SetBy: "kpt",
-		}
-		rw := &kio.LocalPackageReadWriter{
-			PackagePath:           path,
-			KeepReaderAnnotations: false,
-			IncludeSubpackages:    true,
-		}
-		err = kio.Pipeline{
-			Inputs:  []kio.Reader{rw},
-			Filters: []kio.Filter{setter},
-			Outputs: []kio.Writer{rw},
-		}.Execute()
+		err = SetV1AutoSetter(fmt.Sprintf("gcloud.%s", c), v, path)
 		if err != nil {
 			return err
 		}
 
-		setter2 := &settersutil.FieldSetter{
-			Name:          fmt.Sprintf("gcloud.%s", c),
-			Value:         v,
-			SetBy:         "kpt",
-			OpenAPIPath:   filepath.Join(path, "Kptfile"),
-			ResourcesPath: path,
-		}
-		err = kio.Pipeline{
-			Inputs:  []kio.Reader{rw},
-			Filters: []kio.Filter{setter2},
-			Outputs: []kio.Writer{rw},
-		}.Execute()
+		err = SetV2AutoSetter(fmt.Sprintf("gcloud.%s", c), v, path)
 		if err != nil {
 			return err
 		}
+	}
 
-		if c == "core.project" && (setter.Count > 0 || setter2.Count > 0) {
-			// set the projectNumber if we set the projectID
-			projectID = v
-		}
+	return nil
+}
+
+var GetProjectNumberFromProjectID = func(projectID string) (string, error) {
+	gcloudCmd := exec.Command("gcloud",
```

Comment:
How do we handle situations where gcloud is not available?

Reply:
This method is execute iff gcloud.core.project is set which means the gcloud exists. If the gcloud is not installed we continue and not error out. Refer to line 90 comment.

Reply:
👍